docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -159,3 +159,116 @@ Reduction Policies
 For more information about available RAJA reduction policies and guidance
 on which to use with RAJA execution policies, please see 
 :ref:`reducepolicy-label`.
+
+--------------------------------
+Experimental Reduction Interface
+--------------------------------
+
+An experimental reduction interface is now available that hopes to improve
+upon the current reduction model in RAJA. This interface allows ``RAJA::forall``
+to take optional "plugin-like" objects to extend the execution behaviour 
+of a ``RAJA::forall`` execution context.
+
+The new interface passes ``RAJA::expt::Reduce<OP_TYPE>`` objects as direct
+parameters to ``RAJA::forall`` and provides users with thread-local variables
+of the operating data type to be manipulated indside the lambda. This differs 
+from the current reduction model in which ``RAJA::ReduceOP<REDUCE_POL, T>``
+objects are captured by the RAJA lambda and manipulated directly.
+
+
+RAJA::expt::Reduce
+..................
+::
+
+  double rs = 0.0;
+  double rm = 1e100;
+      
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+  RAJA::expt::Reduce<RAJA::operators::plus>(&rs),
+  RAJA::expt::Reduce<RAJA::operators::minimum>(&rm),
+  [=] (int i, double& _rs, double& _rm) {
+    _rs += ...
+    _rm = RAJA_MIN(..., _rm); 
+  }
+  );
+  
+  std::cout << rs ...
+  std::cout << rm ...
+
+* ``RAJA::expt::Reduce`` takes a target variable to write the final result to 
+  (``rs``, ``rm``). The reduction operation will include the initial value of
+  the given target variable.
+* ``RAJA::forall`` requires a corresponding lambda argument to
+  ``RAJA::expt::Reduce`` be passed (``_rs``, ``_rm``). These arguments can be
+  used as the local instance of the target for the reduction operation.
+* The local variables (``_rs, ``_rm``) are initialized with the "identity" of
+  the reduction operation to be performed.
+* A reduction is performed implicitly by the ``RAJA::forall`` across thread 
+  copies of the local variables.
+* Finally, the reduction operation is performed against the original value of 
+  the target and the result of the ``RAJA::forall`` reduction.
+* The final value can be returned simply be referencing the target variable.
+
+RAJA::expt::ValLoc
+..................
+
+RAJA supports ``Loc`` reductions. With this new interface ``Loc`` reductions 
```

Comment:
```suggestion
RAJA supports ``Loc`` reductions, which provide the ability to get a kernel/loop index at which the final reduction value was found. With this new interface ``Loc`` reductions 
```

Reply:
@mdavis36 does this change the semantics of 'loc' reductions? For example, If I initialize the value of a min reduction to -1 and I perform the reduction over an array of positive values, I will get the index of the min value in the array via getLoc(). Is that the same behavior as the current model? I can't remember.

Reply:
If you set the initial value of the target ValLoc to -1 and perform a min reduction on an array of positive integer you will get -1. This is the same behavior as with our current reduction model. You must initialize the value to above the highest value in the array if you wish to find the min loc in the array. Tested with this:
```
  RAJA::TypedRangeSegment<int> testrange(0, 10);
  std::array<int, 10> test = {43,7,6,1,9,12,14,46,78,3};

  std::cout << "\n Running RAJA OLD LOC omp reductions...\n";
  RAJA::ReduceMinLoc< RAJA::omp_reduce, int > vminloc(-1, -1);

  RAJA::forall<RAJA::omp_parallel_for_exec>( testrange,
    [=](RAJA::Index_type i) {

    vminloc.minloc( test[i], i );

  });
  std::cout << "Old min : " << vminloc.get() << "\n";
  std::cout << "Old loc : " << vminloc.getLoc() << "\n";

  std::cout << "\n Running RAJA NEW LOC omp reductions...\n";
  VALLOC_INT new_minloc(-1, -1);

  RAJA::forall<RAJA::omp_parallel_for_exec>(testrange,
    RAJA::expt::Reduce<RAJA::operators::minimum>(&new_minloc),
    [=](int i, VALLOC_INT& _new_minloc) {
      _new_minloc.min(test[i], i);
      //_new_minloc = RAJA_MIN(VALLOC_INT(test[i], i), _new_minloc);
    }
  );

  std::cout << "New min : " << new_minloc.getVal() << "\n";
  std::cout << "New loc : " << new_minloc.getLoc() << "\n";
```
Output:
```
 Running RAJA OLD LOC omp reductions...
Old min : -1
Old loc : -1

 Running RAJA NEW LOC omp reductions...
New min : -1
New loc : -1
```